### DISCUSSION TITLE
ESP32-C6 Support

### DISCUSSION THREAD

**Discussion by Anthael** [arendst/Tasmota#19233] [Ideas]
Hi,

I expect to install tasmota firmware on an ESP32-C6 and I would like to know if anybody try to flash it.

The hardware has following specification:
- manufacturer: expressif 
- product number: ESP32-C6-DEVKITC-1-N8
- driver: CP2102
- other reference : ESP32-C6-WROOM-1 
- Wifi: 802.11 b/g/n/ax (Wi-Fi, WiFi, WLAN), Frequency 2.4 GHz
- 802.15.4 (Thread, ZigBee®)
- Bluetooth 5 2.4GHz 

More info 
https://eu.mouser.com/new/espressif/espressif-esp32-c6-devkitc-1-board/

This board seems quite interesting for all the fonctionnalities ( Zigbee, Thread, BT, Wifi 6 )

Thanks by advance.

**Comment by barbudor** [comment]
As of today C6 is not supported
A few weeks ago, C6 was still not supported by the Arduino framewok on which Tasmota is based.
There seems to be some preliminary support now : https://github.com/espressif/arduino-esp32/issues/7713

But still, this will require some work so do not expect support of C6 in Tasmota before later this year
This is not a priority for the team

**Reply by sfromis** [reply]
And even assuming that some degree of C6 support may (or not) come this year, do not expect it to be including support for all those additional radios, meaning that you may not get much benefit over the standard C3.....

**Reply by Jason2866** [reply]
Supporting the C6 with functions of the C3 will be not so much work. The new features of the C6 are still not supported by Arduino. So we can't do anything here...
The major BIG task is to adopt Tasmota code from Arduino core 2.0.x to 3.0.x.
The underlying IDF version is changed from v4.4x to 5.1.x. This change has a big impact and needs a lot of changes in Tasmota code.
Everyone is invited to do this porting work. There is already a branch for this.
https://github.com/arendst/Tasmota/tree/WiP_Arduino_300
When you start a compile run, after a few lines errors will stop the run ;-)
